src/core/task_mngr.c

```diff
@@ -0,0 +1,195 @@
+#include "task_mngr.h"
+
+
+//this functions are inline because they are run only from task manager at one place.
+void run_systemtasks_preexecute(void);
+void run_systemtasks_postexecute(void);
+void run_calculate_priority(void);
+void run_usertasks(void);
+task_t* find_next_task();
+
+task_t* task_user_active = NULL;
+task_t* task_user_list[] = {[0 ... (TASK_USER_LIST_SIZE-1)] = NULL};
+task_t* task_system_preexecute[] = {[0 ... (TASK_SYSTEM_PREEXECUTE_LIST_SIZE-1)] = NULL};
+task_t* task_system_postexecute[]= {[0 ... (TASK_SYSTEM_POSTEXECUTE_LIST_SIZE-1)] = NULL};
+task_t* task_user_backproc[] = {[0 ... (TASK_USER_BACKPROC_LIST_SIZE-1)] = NULL};
```

Comment:
I suspected that I could rebase on master, but i had to merge and fucked up merge. This should be in Makefile before push. Fixed in last commit